scene/gui/option_button.cpp

```diff
@@ -30,10 +30,44 @@
 
 #include "option_button.h"
 
+#include "core/config/project_settings.h"
 #include "scene/theme/theme_db.h"
 
 static const int NONE_SELECTED = -1;
 
+void OptionButton::gui_input(const Ref<InputEvent> &p_event) {
+	Ref<InputEventKey> k = p_event;
+
+	if (p_event->is_action("ui_cancel", true)) {
+		search_string = "";
+	}
+
+	if (k.is_valid() && k->get_unicode() && k->is_pressed()) {
+		uint64_t now = OS::get_singleton()->get_ticks_msec();
+		uint64_t diff = now - search_time_msec;
+		uint64_t max_interval = uint64_t(GLOBAL_GET("gui/timers/incremental_search_max_interval_msec"));
+		search_time_msec = now;
+
+		if (diff > max_interval) {
+			search_string = "";
+		}
+
+		if (String::chr(k->get_unicode()) != search_string) {
```

Comment:
Why this `!=` test? Wouldn't this prevent searching for repeated chars like `aa`?

Reply:
Hm, no idea why it is here (this code is almost verbatim copy of the equivalent `PopupMenu` code, seem to be introduced in the https://github.com/godotengine/godot/pull/28841), I'll check it.

Reply:
Seems to be working fine without it, so I have removed the check from all 3 places it was used. Still no idea why it was added.

Reply:
See https://github.com/godotengine/godot/pull/102112/files#r1938265579, this is why it was there.